src/services/node-checker.ts

```diff
@@ -0,0 +1,376 @@
+import get from 'lodash/get'
+import { Pocket, Configuration, Node, RelayResponse, PocketAAT, HTTPMethod, RpcError } from '@pokt-network/pocket-js'
+import { blockHexToDecimal } from '../utils/block'
+import { checkEnforcementJSON } from '../utils/enforcements'
+
+export type Check = 'sync-check' | 'chain-check' | 'archival-check'
+
+export type NodeCheckResponse<T> = {
+  node: Node
+  check: Check
+  success: boolean
+  response: string | Error
+  output?: T
+}
+
+export type ChainCheck = {
+  chainID: number
+}
+
+export type SyncCheck = {
+  blockHeight: number
+}
+
+export type ArchivalCheck = {
+  message: string
+}
+
+type ProcessCheck = {
+  success: boolean
+  relayResponse: RelayResponse | Error
+  output: string | number
+}
+
+export class NodeChecker {
+  pocket: Pocket
+  configuration: Configuration | undefined
+
+  constructor(pocket: Pocket, configuration?: Configuration) {
+    this.pocket = pocket
+    this.configuration = configuration || pocket.configuration
+  }
+
+  /**
+   * Returns a decimal representation from a block's hexadecimal response
+   * @param payload object containing the hexadecimal value
+   * @param key key for which the value is stored
+   * @returns decimal representation of the object's response
+   */
+  static parseBlockFromPayload(payload: object, key: string): number {
+    const rawHeight = get(payload, key) || '0'
+
+    return blockHexToDecimal(rawHeight)
+  }
+
+  /**
+   * Perfoms chain check of a node, making sure the chain a node reports to support is the one that is being requested
+   * within the session.
+   * @param node Node to check.
+   * @param data Payload to send to the blockchain, expected to return the blockchain's chainID.
+   * @param blockchainID Blockchain to request data from.
+   * @param aat Pocket Authentication Token object.
+   * @param chainID  blockchain chain's ID to evaluate against.
+   * @param path  optional. Blockchain's path to send the request to.
+   * @returns Response object containing the relay response, request output and boolean
+   * assuring whether the node supports the correct chain or not
+   */
+  async chain(
```

Comment:
Shouldn't we name this `checkChain` or `performChainCheck`?

Reply:
```
const nodeChecker = new nodeChecker(params...)
nodeChecker.chain() 
```

both names seem redundant to me

Reply:
I don't think its redudant at all—and even if it might seem a little bit redundant, having no context is even worse. `nodeChecker.chain()` doesn't tell me much about what this is doing, meanwhile `performChainCheck` does.